common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/text/DaxTextInput.kt

```diff
@@ -52,6 +54,10 @@ interface TextInput {
 
     fun addTextChangedListener(textWatcher: TextWatcher)
     fun removeTextChangedListener(textWatcher: TextWatcher)
+    fun setEndIcon(
+        @DrawableRes endIconRes: Int?,
```

Comment:
I don’t think we should allow this to be nullable

Reply:
it's to remove the icon. Another approach could be to create a different method to `hideEndIcon()` but I think this is simpler.

Reply:
Talk to @malmstein and we agreed on add `removeEndIcon()` so `endIconRes` is not nullable anymore